src/WikiService.Data/RubricRepository.cs

```diff
@@ -1,148 +1,21 @@
-      {
-        dbRubric = dbRubric.Where(
-          rubric =>
-            rubric.Name.Contains(filter.NameIncludeSubstring.ToLower()));
-      }
-
-      if (filter.IsAscendingSort.HasValue)
-      {
-        dbRubric = filter.IsAscendingSort.Value
-          ? dbRubric.OrderBy(rubric => rubric.Name)
-          : dbRubric.OrderByDescending(rubric => rubric.Name);
-      }
-
-      if (filter.IsActive.HasValue)
-      {
-        dbRubric = dbRubric.Where(rubric => rubric.IsActive == filter.IsActive.Value);
-      }
-      else
-      {
-        dbRubric = dbRubric.OrderByDescending(rubric => rubric.CreatedAtUtc);
-      }
-
-      return dbRubric;
-    }
-    
-    private async Task<List<DbRubric>> FindRubricChild(IQueryable<DbRubric> dbRubrics)
-    {
-      List<DbRubric> result = await dbRubrics.ToListAsync();
-      
-      foreach (DbRubric topRubric in result)//todo - to rewrite the next foreach with Any()
-      {
-        foreach (DbRubric rubric in _provider.Rubrics.AsEnumerable())
-        {
-          if (rubric.ParentId == topRubric.Id)
-          {
-            topRubric.HasChild = true;
-            break;
-          }
-        }
-
-        if (!topRubric.HasChild)
-        {
-          foreach (DbArticle article in _provider.Articles.AsEnumerable())
-          {
-            if (article.RubricId == topRubric.Id)
-            {
-              topRubric.HasChild = true;
-              break;
-            }
-          }
-        }
-      }
-
-      return result;
-    }
 
     public RubricRepository(
       IDataProvider provider)
     {
       _provider = provider;
     }
 
-    public async Task<(List<DbRubric> dbRubric, int totalCount)> FindAsync(FindRubricFilter filter)
-    {
-      if (filter is null)
-      {
-        return (null, default);
-      }
-
-      IQueryable<DbRubric> dbRubric = CreateFindPredicates(
-        filter,
-        _provider.Rubrics.AsQueryable().Where(rubric => rubric.ParentId == null));
-
-      return (
-        await FindRubricChild(dbRubric.Skip(filter.SkipCount).Take(filter.TakeCount)),
-        await dbRubric.CountAsync());
-    }
-
-    public async Task<Guid?> CreateAsync(DbRubric dbRubric)
-    {
-      if (dbRubric is null)
-      {
-        return null;
-      }
-
-      _provider.Rubrics.Add(dbRubric);
-      await _provider.SaveAsync();
-
-      return dbRubric.Id;
-    }
-
-    public Task<DbRubric> GetAsync(Guid rubricId)
-    {
-      return _provider.Rubrics.FirstOrDefaultAsync(x => x.Id == rubricId);
-    }
-
-    public async Task<bool> EditAsync(DbRubric dbRubric, JsonPatchDocument<DbRubric> request)
-    {
-      if (dbRubric is null || request is null)
-      {
-        return false;
-      }
-
-      request.ApplyTo(dbRubric);
-      dbRubric.ModifiedBy = _httpContextAccessor.HttpContext.GetUserId();
-      dbRubric.ModifiedAtUtc = DateTime.UtcNow;
-
-      await _provider.SaveAsync();
-
-      return true;
-    }
-
     public async Task<bool> DoesExistAsync(Guid rubricId)
```

Comment:
may be remove DoesExistAsync and DoesRubricNameExistAsync and write business in validators?

Reply:
done